flytekit/clis/sdk_in_container/run.py

```diff
@@ -48,10 +60,51 @@ def convert(self, value, param, ctx) -> typing.Optional[typing.Union[typing.Dict
 class DataframeType(click.ParamType):
     name = "dataframe"
 
-    def convert(self, value, param, ctx) -> pd.DataFrame:
-        if not ctx.obj[REMOTE_KEY]:
+    def __init__(self, input_type: typing.Type[StructuredDataset]):
+        self._sdt = TypeEngine.to_literal_type(input_type)
+
+    def convert(self, value, param, ctx) -> typing.Union[pd.DataFrame, str]:
```

Comment:
will we have to do this for all dataFrame types? This seems a little odd?
like sparkDataFrame etc?

I somehow feel, that we should probably do this on the LiteralType - the click conversion? This universe is much smaller and controlled.